### DISCUSSION TITLE
DLedgerControlller模式可以保证消息一致性吗

### DISCUSSION THREAD

**Discussion by yzq-del** [apache/rocketmq#7691] [General]
DLedgerControlller只是选主吗？如果主从同步中，从数据落后，切主后会不会消息丢了？

**Comment by GenerousMan** [comment]
If you have the allAckInSyncStateSet switch turned on, then no, a replica node that is significantly behind will not be chosen as primary upon startup. 
If a node is in the syncStateSet and then gets elected as primary, there is actually no data loss even if log truncation occurs at this point. This is because this portion of data has not been successfully transmitted - it has not been acknowledged by all the nodes in the syncStateSet.
